sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -162,6 +162,7 @@ trait PrunedScan {
 @DeveloperApi
 trait PrunedFilteredScan {
   def buildScan(requiredColumns: Array[String], filters: Array[Filter]): RDD[Row]
+  def supportPredicate(predicate: Expression): Boolean
```

Comment:
Would be better if we rename this method to `supportFilter` since we're using the term "filter" instead of "predicate" through out the data sources API.